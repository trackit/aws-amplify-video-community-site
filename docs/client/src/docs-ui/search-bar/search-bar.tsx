import {Component, Host, h, Build, Element} from "@stencil/core";
import {searchStyle} from "./search-bar.style";
import {transformData} from "../../utils/transform-search-data";
import {setSearchQuery, trackSearchQuery} from "../../utils/track";
import {
  ALGOLIA_API_KEY,
  ALGOLIA_INDEX_NAME,
  UNINITIALIZED_SEARCH_INPUT_SELECTOR,
} from "../../constants/algolia";

@Component({tag: "docs-search-bar", shadow: false})
export class DocsSearchBar {
  @Element() element: HTMLElement;

  initDocSearch() {
    if (
      Build.isBrowser &&
      document.querySelector(UNINITIALIZED_SEARCH_INPUT_SELECTOR)
    ) {
      docsearch({
        apiKey: ALGOLIA_API_KEY,
        indexName: ALGOLIA_INDEX_NAME,
        inputSelector: UNINITIALIZED_SEARCH_INPUT_SELECTOR,
        debug: false,
        queryHook: setSearchQuery,
        handleSelected: trackSearchQuery,
        transformData,
        algoliaOptions: {
          hitsPerPage: 10,
        },
      });
    }
  }

  componentDidLoad() {
    this.initDocSearch();
  }

  render() {
    return (
      <Host class={searchStyle}>
        <div>
          <div>
            <input
              id="amplify-docs-search-input"
              class="three-dee-effect"
              type="search"
              placeholder="Search"
            />
            <img src="/assets/search.svg" alt="search" />
          </div>
        </div>
      </Host>
    );
  }
}
