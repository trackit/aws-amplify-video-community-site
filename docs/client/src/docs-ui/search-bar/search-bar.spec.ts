import {DocsSearchBar} from "./search-bar";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-search-bar", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsSearchBar],
            html: `<docs-search-bar />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
