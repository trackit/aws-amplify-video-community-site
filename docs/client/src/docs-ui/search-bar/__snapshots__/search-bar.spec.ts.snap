// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-search-bar Render logic should render 1`] = `
<docs-search-bar class="css-j5bz1e">
  <div>
    <div>
      <input class="three-dee-effect" id="amplify-docs-search-input" placeholder="Search" type="search">
      <img alt="search" src="/assets/search.svg">
    </div>
  </div>
</docs-search-bar>
`;
